tools/pack_model_artifacts.py

```diff
@@ -0,0 +1,83 @@
+                   BUILTIN_TASK_NODE_REGRESSION,
+                   BUILTIN_TASK_EDGE_CLASSIFICATION,
+                   BUILTIN_TASK_EDGE_REGRESSION]
+
+def main(args):
+    """ Wrap model artifacts and entry point file into a compressed tar file.
+    """
+    # chose the target entry point file according to task
+    entry_base_path = 'python/graphstorm/sagemaker/deploy/'
+    if args.gml_task == BUILTIN_TASK_NODE_CLASSIFICATION:
+        path_to_entry = os.path.join(args.graphstorm_home_folder, entry_base_path,
+                                     'node_classification_entry_point.py')
+    elif args.gml_task == BUILTIN_TASK_NODE_REGRESSION:
+        path_to_entry = os.path.join(args.graphstorm_home_folder, entry_base_path,
+                                     'node_regression_entry_point.py')
+    elif args.gml_task == BUILTIN_TASK_EDGE_CLASSIFICATION:
+        path_to_entry = os.path.join(args.graphstorm_home_folder, entry_base_path,
+                                     'edge_classification_entry_point.py')
+    elif args.gml_task == BUILTIN_TASK_EDGE_REGRESSION:
+        path_to_entry = os.path.join(args.graphstorm_home_folder, entry_base_path,
+                                     'edge_regression_entry_point.py')
+    else:
+        raise NotImplementedError(f'Not supported GML task {args.gml_task}.')
+
+    output_file = wrap_model_artifacts(args.model_file_path,
+                                        args.model_config_yaml_paath,
+                                        args.graph_metadata_json_path,
+                                        path_to_entry, wrap_name='model',
+                                        output_path=args.output_folder)
+    print(f'Successfully pack GraphStorm model artifacts into {output_file}.')
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser('Package GraphStorm model artifacts.')
+    parser.add_argument('--model-file-path', type=str, required=True,
+                        help='The file path of trained GraphStorm model. By default, ' + \
+                             'GraphStorm saves model parameters into a binary file, named ' + \
+                             '\'model.bin\' file.')
+    parser.add_argument('--model-config-yaml-path', type=str, required=True,
+                        help='The file path of model configuration YAML file. By default, ' + \
+                             'GraphStorm generates a new YAML file that stores all model ' + \
+                             'configurations during a training.')
+    parser.add_argument('--graph-metadata-json-path', type=str, required=True,
+                        help='The file path of graph metadata JSON file. By default, ' + \
+                             'GraphStorm generates a new JSON file that stores all graph ' + \
+                             'metadata during a graph construction.')
+    parser.add_argument('--graphstorm-home-folder', type=str, required=True,
```

Comment:
You assume the user clones the graphstorm repo on its local disk.
But users do not necessarily clone the repo. They only need to download the entry_point file or they may customize their own entry_point.

Reply:
+1 this is related to the above comment on where this file should live as well.

Reply:
See the reply of the next comment.